tap_redshift/__init__.py

```diff
@@ -366,35 +366,48 @@ def sync_table(connection, catalog_entry, state):
         elif replication_key is not None:
             select += ' ORDER BY {} ASC'.format(replication_key)
 
-        time_extracted = utils.now()
-        query_string = cursor.mogrify(select, params)
-        LOGGER.info('Running {}'.format(query_string))
-        cursor.execute(select, params)
-        row = cursor.fetchone()
-        rows_saved = 0
+        if limit:
+            select += ' LIMIT %(limit)s OFFSET %(offset)s'
```

Comment:
Do we need to make sure the query results are ordered exactly the same way each time?

Reply:
It is already ordering [here](https://github.com/Automattic/tap-redshift/blob/d542e736f0f06fea47b75aea20543e4bceb8cb7c/tap_redshift/__init__.py#L367) (only when the replication_key is defined).

Reply:
Gotcha, I'm a little worried about ordering just on `batch_timestamp` because many records will have the same timestamp (6 hour increments & nulls). So the order of rows within `batch_timestamp` could change every time a query is a run for each LIMIT/OFFSET iteration which I think is how it works, right?

Reply:
That's an interesting point, not sure if it keeps the order for rows with the same timestamp. I need to investigate it.

Reply:
Ok, I did a small test in redshift interface with the query:
```sql
SELECT * from sdp.stripe.connected_account_charges_metadata 
order by batch_timestamp ASC NULLS FIRST LIMIT 100 OFFSET 100
``` 
```sql
# changed the limit to avoid the cache result
SELECT * from sdp.stripe.connected_account_charges_metadata 
order by batch_timestamp ASC NULLS FIRST LIMIT 10 OFFSET 100
``` 
To get only null values and compere the first page and there rows didn't match 😨

Reply:
I'm checking if we can send more columns to order by in the metadata.

Reply:
got it, thanks for checking! Let me know if there's anything I can help with. I think sending more columns is the way to go. Wondering how columns should be chosen, though. If we were to generalize this to any table, we could maybe add an ORDER BY clause followed by an integer sequence up to the number of columns getting synced? For example, `ORDER BY 1,2,3,4,5` for a table with 5 columns

Reply:
Just saw this message @ytanaka6174, sorry! I've added the order `extra-order-by-columns` in metadata that you can add how many columns you want (but the first will be always the replication key).